src/platforms/android/server/display.cpp

```diff
@@ -359,9 +359,9 @@ bool mga::Display::apply_if_configuration_preserves_display_buffers(
     if (!config.external().connected || displays.display_present(mga::DisplayName::external))
```

Comment:
To detect if the buffers should be preserved, the condition should read:
```c++
        if (config.external().connected || !displays.display_present(mga::DisplayName::external))
```
i.e. preserve the buffer if the external display is connected, or it's not but we've already dealt with it.


But I would rather change the whole function to read:
```c++
        if (!config.external().connected && displays.display_present(mga::DisplayName::external))
            return false;

        configured_locked(conf, lock);
        return true;
```
i.e. detect when _not_ to preserve the buffer instead.

Reply:
Yeah i think this is indeed correct, we should return if previous display has been disconnected and call configured_locked otherwise.  

So the *root* of the problem is that we need to restart the compositor when a display has been disconnected. 

@RAOF is this correct?